### PR TITLE
add extra dirname to get correct package_name

### LINKED ISSUE #120: The identified package name in check_deps is wrong
To reproduce, `Pkg.add("SpecialFunctions")` then take a look at the generated `deps.jl` file. Mine looks like
```julia
## This file autogenerated by BinaryProvider.write_deps_file().
## Do not edit.
##
## Include this file within your main top-level source, and call
## `check_deps()` from within your module's `__init__()` method

if isdefined((@static VERSION < v"0.7.0-DEV.484" ? current_module() : @__MODULE__), :Compat)
    import Compat.Libdl
elseif VERSION >= v"0.7.0-DEV.3382"
    import Libdl
end
const openspecfun = joinpath(dirname(@__FILE__), "usr/lib/libopenspecfun.1.3.dylib")
function check_deps()
    global openspecfun
    if !isfile(openspecfun)
        error("$(openspecfun) does not exist, Please re-run Pkg.build(\"KvXoO\"), and restart Julia.")
    end

    if Libdl.dlopen_e(openspecfun) in (C_NULL, nothing)
        error("$(openspecfun) cannot be opened, Please re-run Pkg.build(\"KvXoO\"), and restart Julia.")
    end

end
```
Note in particular
> Please re-run `Pkg.build(\"KvXoO\")`

This is due to this line:
https://github.com/JuliaPackaging/BinaryProvider.jl/blob/d0e78699605168ffa8606cb7a0f0ed21cf9999f3/src/Products.jl#L398
which is not quite right. In my case the path is `/Users/alex/.julia/packages/SpecialFunctions/KvXoO/deps/deps.jl`.

### PR DESCRIPTION
**Pull request by simonbyrne**
Hopefully fix #120.

### CODE REVIEW

File: src/Products.jl [hunk 1]
```diff
@@ -395,8 +395,11 @@ function write_deps_file(depsjl_path::AbstractString, products::Vector{P};
     escape_path = path -> replace(path, "\\" => "\\\\")
 
     # Grab the package name as the name of the top-level directory of a package
-    package_name = basename(dirname(dirname(depsjl_path)))
-
+    if VERSION < v"0.7.0-"
```
Comment (ararslan) [inline src/Products.jl]: Is the `VERSION` check necessary, since we currently require 0.7 at a minimum?

**Comment by fredrikekre** [comment]
You can not get the package name from the file path in general. In particular this will not work for packages that are `dev`ed.

File: src/Products.jl (hunk 1 above)
Comment (simonbyrne) [reply src/Products.jl]: actually, it occurs to me that this won't work for `dev`ed packages either.

**Comment by staticfloat** [comment]
It Works

**Pull request merged**